bliss/models/galaxy_net.py

```diff
@@ -139,43 +140,68 @@ def validation_step(self, batch, batch_idx):  # pylint: disable=unused-argument
         loss = self.get_loss(images, recon_mean)
 
         # metrics
+        # FIXME: change to val/loss so it's automatically grouped in tensorboard
         self.log("val_loss", loss)
         residuals = (images - recon_mean) / torch.sqrt(images)
         self.log("val_max_residual", residuals.abs().max())
         return {"images": images, "recon_mean": recon_mean}
 
     def validation_epoch_end(self, outputs):
-        images = outputs[0]["images"][:10]
-        recon_mean = outputs[0]["recon_mean"][:10]
+        images = torch.cat([x["images"] for x in outputs])
+        recon_mean = torch.cat([x["recon_mean"] for x in outputs])
         fig = self.plot_reconstruction(images, recon_mean)
         if self.logger:
             self.logger.experiment.add_figure(f"Images {self.current_epoch}", fig)
 
     def plot_reconstruction(self, images, recon_mean):
-        # only plot i band if available, otherwise the highest band given.
+
+        # 1. only plot i band if available, otherwise the highest band given.
+        # 2. plot the `num_examples` of images with the largest average residual
+        #    across all batches in the last epoch
+        # 3. residual color range (`vmin`, `vmax`) are fixed as across all samples
+        #    (same across all rows in the subplot grid)
+        # 4. image and recon_mean are fixed in each sample
+        #    (same across each row in the subplot grid)
+
         assert images.size(0) >= 10
         num_examples = 10
         num_cols = 3
         residuals = (images - recon_mean) / torch.sqrt(images)
+        large_residuals_idx = residuals.mean(dim=(1, 2, 3)).argsort(descending=True)[:num_examples]
+        images = images[large_residuals_idx]
+        recon_mean = recon_mean[large_residuals_idx]
+        residuals = residuals[large_residuals_idx]
+
+        residual_vmax = torch.ceil(residuals.max().cpu()).numpy()
+        residual_vmin = torch.floor(residuals.min().cpu()).numpy()
         plt.ioff()
 
         fig = plt.figure(figsize=(10, 25))
 
         for i in range(num_examples):
+            image = images[i, 0].data.cpu()
+            recon = recon_mean[i, 0].data.cpu()
+            vmax = torch.ceil(torch.max(image.max(), recon.min()))
+            vmin = torch.floor(torch.min(image.min(), recon.min()))
 
             plt.subplot(num_examples, num_cols, num_cols * i + 1)
             plt.title("images")
-            plt.imshow(images[i, 0].data.cpu().numpy())
+            plt.imshow(image.numpy(), interpolation=None, vmin=vmin, vmax=vmax)
             plt.colorbar()
 
             plt.subplot(num_examples, num_cols, num_cols * i + 2)
             plt.title("recon_mean")
-            plt.imshow(recon_mean[i, 0].data.cpu().numpy())
+            plt.imshow(recon.numpy(), interpolation=None, vmin=vmin, vmax=vmax)
```

Comment:
Thanks the `interpolation=None` is actually important. I think this is done by default but it's good to be explicit here.

Reply:
Not sure this was set in a global config somewhere else. I thought `interpolation='antialiased'` was the default setting. If this is the case, is there other places I should look into and maybe change those as well?